### PR TITLE
Save previous calculations of the user in the storage

### PR DESCRIPTION
**Pull request by SleekMutt**
https://github.com/user-attachments/assets/ad80f149-7a3c-4165-8e9d-0a611ec4ecc4

solves #990 

## Code reviewers

- [ ] @loqimean 

## Summary of issue

Users calculations must be saved in the storage

## Summary of change

Added feature to save previous results in the session storage

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: app/controllers/calculators_controller.rb [hunk 1]
```diff
@@ -22,6 +22,9 @@ def calculate
 
     @results = Calculators::CalculationService.new(@calculator, params[:inputs]).perform
 
+    session[:calculation_results]                 ||= {}
```
Comment (loqimean) [inline app/controllers/calculators_controller.rb:25]: тести))

File: app/views/calculators/show.html.erb [hunk 2]
```diff
@@ -10,12 +10,16 @@
...
+      <br>
     <% end %>
     <div class="mt-6 form-actions">
       <%= form.submit "Calculate", class: "bg-blue-500 text-white font-semibold px-4 py-2 rounded hover:bg-blue-700" %>
     </div>
   <% end %>
-
-  <%= turbo_frame_tag "calc-results" %>
+  <%= turbo_frame_tag "calc-results" do %>
+    <% if session.dig(:calculation_results, @calculator.slug)&.presence %>
```
Comment (loqimean) [inline app/views/calculators/show.html.erb:21]:
```suggestion
    <% if session.dig(:calculation_results, @calculator.slug).present?%>

```

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/calculators/show.html.erb [hunk 3]
```diff
@@ -10,12 +10,16 @@
...
     <% end %>
     <div class="mt-6 form-actions">
       <%= form.submit "Calculate", class: "bg-blue-500 text-white font-semibold px-4 py-2 rounded hover:bg-blue-700" %>
     </div>
   <% end %>
-
-  <%= turbo_frame_tag "calc-results" %>
+  <%= turbo_frame_tag "calc-results" do %>
+    <% if session.dig(:calculation_results, @calculator.slug).present? %>
+      <%= render "calculators/partials/calculation_results", results: session[:calculation_results][@calculator.slug].map(&:symbolize_keys) %>
```
Comment (loqimean) [inline app/views/calculators/show.html.erb:22]: 1. use `with_indifferent_access`
2. use `dig`

**Review by loqimean — CHANGES_REQUESTED** [review]

**Comment by loqimean** [comment]
1. solve conflicts
2. request team review, as it is required

**Review by ktpe — APPROVED** [review]

**Review by loqimean — APPROVED** [review]

**Pull request merged**